src/basic/string-util.h

```diff
@@ -130,7 +130,7 @@ static inline bool _pure_ in_charset(const char *s, const char* charset) {
 }
 
 static inline bool char_is_cc(char p) {
-        return (p >= 0 && p < ' ') || p == 127;
+        return (uint8_t) p < ' ' || p == 127;
```

Comment:
it might make sense to cast both the lhs *and* the rhs of the `<` comparison. i mean, according to the C standard comparing an unsigned with a signed variable means both are cast to unsigned, so this fix works correctly, but it takes quite some knowledge of C to know that these are the rules. (I had to look it up)